tasks/au/data.py

```diff
@@ -46,15 +45,21 @@ class DownloadData(DownloadUnzipTask):
     state = Parameter()
     header = Parameter()
 
+    def version(self):
+        return 1
+
+    def requires(self):
+        return RepoFile(resource_id=self.task_id,
+                        version=self.version(),
+                        url=URL.format(year=self.year,
+                                       profile=self.profile,
+                                       resolution=self.resolution,
+                                       state=self.state,
+                                       header=self.header))
+
     def download(self):
-        urllib.request.urlretrieve(url=URL.format(
-                               year=self.year,
-                               profile=self.profile,
-                               resolution=self.resolution,
-                               state=self.state,
-                               header=self.header
-                           ),
-                           filename=self.output().path + '.zip')
+        self.output().makedirs()
```

Comment:
As `copyfile` receives the output file and recreating the needed directory structure if needed is a safe operation, I'd consider moving the equivalent to `self.output().makedirs()` into `copyfile`.

Reply:
`copyfile` is a function from `shutil` (included in _The Python Standard Library_)

Reply:
Well, I'd wrap it in a `copyfile_makedirs` function 😅 (sorry about not realizing that it's an imported function, I still don't get used to that :_) ).